docs: improve docs of elements using DataItem interface (#948)
Refinitiv/refinitiv-ui: documents/src/pages/elements/list.md

Reviewer: "Usage" and "Data Property Interface" look duplicate.
Let's combine them into "Usage" similar to other elements with this order.
```md
### Usage
The easiest way to populate the list...

<-- javascript code snippet -->

The `data` property of the `ef-list` use...
```

Author: Updated by removing `Data property interface` header.

## Usage

> The easiest way to populate the list is to pass an array of data items to `data` property. Items must adhere to the [Data Item](./custom-components/utils/data-management#data-item) interface.

Reviewer: ```suggestion
The easiest way to populate the list is to pass an array of data items to `data` property. Items must adhere to [DataItem](./custom-components/utils/data-management#data-item).
```

}
```

> The `data` property of the `ef-list` use the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) type for its data items. Each of this item defaults to type `ItemData`. It could also be anything extended from [DataItem](./custom-components/utils/data-management#data-item).

Reviewer: ```suggestion
The `data` property of the `ef-list` uses the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) type for its data items. Each of this item defaults to type `ItemData`. It could also be anything extended from [DataItem](./custom-components/utils/data-management#data-item).
```

## Data property interface

> The `data` property of the `ef-list` use the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) interface that extend from [Data Item](./custom-components/utils/data-management#data-item) interface.

Reviewer 2: ```suggestion
The `ef-list` use the `data` property  `ListData` interface that extend from [Data Item](./custom-components/utils/data-management#data-item) interface.
```

> The `data` property of the `ef-list` use the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) interface that extend from [Data Item](./custom-components/utils/data-management#data-item) interface.

Reviewer 3: This statement is wrong.

Author: Could you tell me why? I see ListData uses ItemData by default. ItemData extends DataItem.

> The `data` property of the `ef-list` use the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) interface for its data items. Each item defaults to type ItemData, it can be anything extended from [DataItem](./custom-components/utils/data-management#data-item).

Reviewer 3: ```suggestion
The `data` property of the `ef-list` use the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) interface for its data items. Each item defaults to type `ItemData`, it can be anything extended from [DataItem](./custom-components/utils/data-management#data-item).
```

> The `data` property of the `ef-list` use the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) interface for its data items. Each item defaults to type `ItemData`, it can be anything extended from [DataItem](./custom-components/utils/data-management#data-item).

Reviewer 3: ```suggestion
The `data` property of the `ef-list` use the [ListData](https://github.com/Refinitiv/refinitiv-ui/blob/v7/packages/elements/src/list/helpers/types.ts) interface for its data items. Each item defaults to type `ItemData`, otherwise it can be any interface that extended from [DataItem](./custom-components/utils/data-management#data-item).
```

Author: Updated by following P'Ole suggestions of comboBox pattern.
